pandas/core/indexes/datetimelike.py

```diff
@@ -530,6 +531,83 @@ def isin(self, values):
 
         return algorithms.isin(self.asi8, values.asi8)
 
+    def intersection(self, other, sort=False):
+        """
```

Comment:
instead of a combined doc-string here, put it in the sub-classes (and call super). @jbrockmendel might have a better way (shared docs) of how we do this.

Reply:
Is that worth it just to avoid the "DatetimeIndex or TimedeltaIndex" ?

Reply:
yes, we don't have combined doc-strings elsewhere so why start here